jvm/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -70,9 +72,18 @@ case class TransformPreOverrides() extends Rule[SparkPlan] {
         new BatchScanExecTransformer(plan.output, plan.scan, newPartitionFilters)
       case plan: FileSourceScanExec =>
         logDebug(s"Columnar Processing for ${plan.getClass} is currently supported.")
+        val needToFillAttr = conf.getConfString(GlutenConfig.GLUTEN_FILL_ATTRIBUTE_FOR_SCAN,
+            "true").toBoolean
+        val realOutput = if (plan.output.size == 0 && needToFillAttr) {
```

Comment:
It's better to use SPI API to support this future for ClickHouse Backend, Velox Backend does not need this.